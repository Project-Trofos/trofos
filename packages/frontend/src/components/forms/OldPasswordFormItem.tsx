import React from 'react';
import { Form, Input } from 'antd';

export default function OldPasswordFormItem(): JSX.Element {
  return (
    <Form.Item
      label="Old Password"
      name="oldPassword"
      rules={[
        {
          required: true,
          message: 'Please enter your old password',
        },
      ]}
    >
      <Input type="password" />
    </Form.Item>
  );
}
