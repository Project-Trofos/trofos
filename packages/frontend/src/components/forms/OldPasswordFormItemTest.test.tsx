import React from 'react';
import { render, screen } from '@testing-library/react';
import FormWrapper from './__testutils__/FormRenderHelper.test';
import OldPasswordFormItem from './OldPasswordFormItem';

describe('OldPasswordFormItem test', () => {
  function renderForm() {
    return render(
      <FormWrapper>
        <OldPasswordFormItem />
      </FormWrapper>,
    );
  }

  describe('when rendering', () => {
    it('should have an old password field', () => {
      const { baseElement } = renderForm();
      expect(baseElement).toMatchSnapshot();

      expect(screen.getByLabelText(/old password/i)).toBeInTheDocument();
    });
  });
});
