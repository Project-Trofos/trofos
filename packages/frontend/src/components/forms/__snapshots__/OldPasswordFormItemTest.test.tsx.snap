// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`OldPasswordFormItem test > when rendering > should have an old password field 1`] = `
<body>
  <div>
    <form
      class="ant-form ant-form-horizontal"
    >
      <div
        class="ant-form-item"
      >
        <div
          class="ant-row ant-form-item-row"
        >
          <div
            class="ant-col ant-form-item-label"
          >
            <label
              class="ant-form-item-required"
              for="oldPassword"
              title="Old Password"
            >
              Old Password
            </label>
          </div>
          <div
            class="ant-col ant-form-item-control"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <input
                  aria-required="true"
                  class="ant-input ant-input-outlined"
                  id="oldPassword"
                  type="password"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <button
        class="ant-btn ant-btn-default ant-btn-color-default ant-btn-variant-outlined"
        type="submit"
      >
        <span>
          submit
        </span>
      </button>
    </form>
  </div>
</body>
`;
